opal_common/authentication/authz.py

```diff
@@ -16,4 +17,15 @@ def require_peer_type(authenticator: JWTAuthenticator, claims: JWTClaims, requir
         raise Unauthorized(description=f"Invalid 'peer_type' claim for OPAL jwt token: {peer_type}")
 
     if type != required_type:
-        raise Unauthorized(description=f"Incorrect 'peer_type' claim for OPAL jwt token: {str(type)}, expected: {str(required_type)}")
\ No newline at end of file
+        raise Unauthorized(description=f"Incorrect 'peer_type' claim for OPAL jwt token: {str(type)}, expected: {str(required_type)}")
+
+
+def restrict_optional_topics_to_publish(authenticator: JWTAuthenticator, claims: JWTClaims, update:DataUpdate):
+    if not authenticator.enabled:
+        return
+
+    if not "permitted_topics" in claims:
+        return
+    
+    if any(set(entry.topics).difference(claims["permitted_topics"]) for entry in update.entries):
+        raise Unauthorized(description=f"Invalid 'topics' to publish")
```

Comment:
Let's replace this with the other option you mentioned - so that the error to the client is more clear:
```
Cannot publish to restricted topic: `bla`
```